openeo/metadata.py

```diff
@@ -820,3 +748,181 @@ def get_temporal_dimension(self, stac_obj: pystac.STACObject) -> Union[TemporalD
             if len(temporal_dims) == 1:
                 name, extent = temporal_dims[0]
                 return TemporalDimension(name=name, extent=extent)
+
+    def _band_from_eo_bands_metadata(self, band: Union[dict, pystac.extensions.eo.Band]) -> Band:
+        """Construct band from metadata dict in eo v1.1 style"""
+        if isinstance(band, pystac.extensions.eo.Band):
+            return Band(
+                name=band.name,
+                common_name=band.common_name,
+                wavelength_um=band.center_wavelength,
+            )
+        elif isinstance(band, dict) and "name" in band:
+            return Band(
+                name=band["name"],
+                common_name=band.get("common_name"),
+                wavelength_um=band.get("center_wavelength"),
+            )
+        else:
+            raise ValueError(band)
+
+    def _band_from_common_bands_metadata(self, data: dict) -> Band:
+        """Construct band from metadata dict in STAC 1.1 + eo v2 style metadata"""
+        # TODO: also support pystac wrapper when available (pystac v2?)
+        return Band(
+            name=data["name"],
+            common_name=data.get("eo:common_name"),
+            wavelength_um=data.get("eo:center_wavelength"),
+        )
+
+    def bands_from_stac_object(self, obj: Union[pystac.STACObject, pystac.Asset]) -> _BandList:
+        # Note: first check for Collection, as it is a subclass of Catalog
+        if isinstance(obj, pystac.Collection):
+            return self.bands_from_stac_collection(collection=obj)
+        elif isinstance(obj, pystac.Catalog):
+            return self.bands_from_stac_catalog(catalog=obj)
+        elif isinstance(obj, pystac.Item):
+            return self.bands_from_stac_item(item=obj)
+        elif isinstance(obj, pystac.Asset):
+            return self.bands_from_stac_asset(asset=obj)
+        else:
+            raise ValueError(f"Unsupported STAC object: {obj!r}")
+
+    def bands_from_stac_catalog(self, catalog: pystac.Catalog) -> _BandList:
+        # TODO: "eo:bands" vs "bands" priority based on STAC and EO extension version information
+        summaries = catalog.extra_fields.get("summaries", {})
+        self._warn(f"bands_from_stac_catalog with {summaries.keys()=} (which is non-standard)")
+        if "eo:bands" in summaries:
+            return _BandList(self._band_from_eo_bands_metadata(b) for b in summaries["eo:bands"])
+        elif "bands" in summaries:
```

Comment:
warn if no STAC 1.1?

Reply:
hmm apparently it's not possible to inspect the original `stac_version` at this point: https://github.com/stac-utils/pystac/issues/1562
so that's a wontfix at the moment unfortunatly